src/inmanta/server/services/resourceservice.py

```diff
@@ -1038,3 +1039,62 @@ async def versioned_resource_details(
         if not resource:
             raise NotFound("The resource with the given id does not exist")
         return resource
+
+    @handle(methods_v2.unmanaged_resource_create)
+    async def unmanaged_resource_create(
+        self, env: data.Environment, unmanaged_resource_id: str, values: Dict[str, str]
+    ) -> None:
+        unmanaged_resource = UnmanagedResource(unmanaged_resource_id=unmanaged_resource_id, values=values)
+        await data.UnmanagedResource(
+            environment=env, unmanaged_resource_id=unmanaged_resource.unmanaged_resource_id, values=unmanaged_resource.values
+        ).insert()
+
+    @handle(methods_v2.unmanaged_resource_create_batch)
+    async def unmanaged_resources_create_batch(
+        self, env: data.Environment, unmanaged_resources: List[UnmanagedResource]
+    ) -> None:
+        resources: List[data.UnmanagedResource] = []
+        for res in unmanaged_resources:
+            unmanaged_resource = data.UnmanagedResource(
+                environment=env,
+                unmanaged_resource_id=res.unmanaged_resource_id,
+                values=res.values,
+            )
```

Comment:
I would add a `to_dao` method to the `UnmanagedResource` class. Then we can reuse that logic across the different API endpoints.